### PR TITLE
Permit ns none

### PR DESCRIPTION
**Pull request by bphunter1972**
I have simple needs. Adding namespaces are not strictly necessary, and in fact would have borked the downstream tool my XML files were going to.

It seemed to me that making ns optional is a reasonable alternative, but I'm not an expert on XML.

### CODE REVIEW

**Comment by tobywf** [comment]
Namespaces are in a way already optional, in that it is already possible to set `__ns__ = None`. This is for XML documents without namespaces:

```python
from xml_dataclasses import load, xml_dataclass
from lxml import etree

@xml_dataclass
class Foo:
    __ns__ = None

el = etree.fromstring('<?xml version="1.0"?>\n<foo />\n')
load(Foo, el, "foo")
```

This will also protect against loading an element without a namespace where one was expected, or loading an element with a namespace where none was expected (see https://github.com/tobywf/xml_dataclasses/blob/30a233defa94c24ead5866c16106719969c5d5d4/tests/load_test.py#L50 and https://github.com/tobywf/xml_dataclasses/blob/30a233defa94c24ead5866c16106719969c5d5d4/tests/load_test.py#L64).

Does that work for your use-case?

It might be more typing, but I think it's good to be explicit about the namespace. This is because `xml_dataclasses` relies on `lxml`, and to resolve tags properly, a namespace is often required. I'm hesitant to make it so that `__ns__` can be entirely omitted. It's possible then to forget to declare `__ns__`, get the "default" of `None`/no namespace, and have the parsing not work.

**Comment by bphunter1972** [comment]
Sure, I actually don't have a problem with declaring the __ns__. 
Mostly, this change had to do with setting `nsmap` to `None` in the `dump()` command:

`
Argument 3 to "dump" has incompatible type "None"; expected "Mapping[Optional[str], str]"
     lxml_el_out = dump(my_list, "my_list", None)
`

The code will run, but mypy doesn't accept this. The change to the type definition of NsMap is what I really need. Using Optional instead of Mapping is the key here.

**Comment by bphunter1972** [comment]
I updated my pull request to focus on just those changes.

**Comment by tobywf** [comment]
Sorry, I still see the old changes in the PR. BTW, I believe `dump(my_list, "my_list", {})` works fine for now. I do think allowing `nsmap` to be `None` is fine/good.

**Comment by bphunter1972** [comment]
Sorry, fixed that.

File: src/xml_dataclasses/resolve_types.py [hunk 1]
```diff
@@ -28,8 +28,8 @@
 from .lxml_utils import format_ns
 
 NoneType: Type[Any] = type(None)
-# Union here is a hack to make this a valid alias
-NsMap = Union[Mapping[Optional[str], str]]
+
+NsMap = Optional[Mapping[Optional[str], str]]
 
 
 class XmlDataclass:
```
Comment (tobywf) [inline src/xml_dataclasses/resolve_types.py:35]: `__nsmap__: Optional[NsMap]` could be `__nsmap__: NsMap` with this I think?

**Comment by tobywf** [comment]
I've incorporated this fix into the main branch. Thanks for cutting the PR!

**Pull request closed without merging**